django_irods/storage.py

```diff
@@ -71,23 +72,41 @@ def zipup(self, in_name, out_name):
         :param out_name: the output zipped file name
         :return: None
         """
-        in_bucket, in_path = bucket_and_name(in_name)
+        def chunk_request(zip_archive_file, bucket, key):
+            chunk_size = 1024 * 1024 * 256  # 256MB
+            object_attrs = self.connection.meta.client.get_object_attributes(Bucket=bucket, Key=key,
+                                                                             ObjectAttributes=["ObjectSize"])
+            object_size = object_attrs.get("ObjectSize")
+
+            chunk_start = 0
+            chunk_end = chunk_start + chunk_size - 1
+
+            while chunk_start < object_size:
+                # Read specific byte range from file as a chunk. We do this because AWS server times out and sends
+                # empty chunks when streaming the entire file.
+                if body := self.connection.meta.client.get_object(
+                    Bucket=bucket, Key=key, Range=f"bytes={chunk_start}-{chunk_end}"
+                ).get("Body"):
```

Comment:
I guess we are confident that range headers are passed to minio appropriately? You tested on beta so I think it is all set

Reply:
yeah, I tested it on beta with the harvey dataset. Also, the unit test use the range headers even though they are very small in comparison.